src/interactive-auth.js

```
@@ -168,15 +257,64 @@ InteractiveAuth.prototype = {
         if (!nextStage) {
             throw new Error("No incomplete flows from the server");
         }
+        if (nextStage == this._currentStage) {
```

won't this swallow errors from (eg) password login?